tracer/src/Datadog.Trace/Logging/LogsInjection/LibLogScopeEventSubscriber.cs

```diff
@@ -126,23 +113,29 @@ public LibLogScopeEventSubscriber(IScopeManager scopeManager, string defaultServ
                 if (_logProvider is ILogProviderWithEnricher logProvider)
                 {
                     var enricher = logProvider.CreateEnricher();
-                    enricher.Initialize(scopeManager, defaultServiceName, version, env);
+                    enricher.Initialize(defaultServiceName, version, env);
                     _logEnricher = enricher;
 
                     _scopeManager.TraceStarted += RegisterLogEnricher;
                     _scopeManager.TraceEnded += ClearLogEnricher;
                 }
-                else if (_logProvider is SerilogLogProvider)
+                else if (_logProvider is FallbackNLogLogProvider)
                 {
-                    // Do not set default values for Serilog because it is unsafe to set
-                    // except at the application startup, but this would require auto-instrumentation
-                    _scopeManager.SpanOpened += StackOnSpanOpened;
-                    _scopeManager.SpanClosed += StackOnSpanClosed;
+                    _scopeManager.DistributedSpanActivated += MapOnDistributedSpanActivated;
+                    _scopeManager.SpanDeactivated += MapOnSpanDeactivated;
                 }
                 else
                 {
-                    _scopeManager.SpanActivated += MapOnSpanActivated;
-                    _scopeManager.TraceEnded += MapOnTraceEnded;
+                    // Do not subcribe to events. This is a proactive step to avoid issues
+                    // when the manual and automatic versions do not match.
+                    //
+                    // The issue is that two separate Datadog.Trace assemblies will be loaded
+                    // and each will create their own LibLogScopeEventSubscriber to modify
+                    // the static state of the logging libraries. Each has an instance-level stack to
+                    // set/unset properties (using the IDisposable pattern), but the two sides
+                    // cannot communicate so they may clear state out of the order in which it was set.
+                    //
+                    // Thus, we will be proactive and disable usage for unknown log providers.
```

Comment:
Good catch - I wonder if this scenario has been at play in some of the weird log injection issues we've seen with "partial" injection etc

Reply:
My thinking exactly. Also, the LibLogScopeEventSubscriber used to have a `ConcurrentStack` instead of an `AsyncLocal` to keep track of its operations, so I feel like that also could contribute to incorrect state.